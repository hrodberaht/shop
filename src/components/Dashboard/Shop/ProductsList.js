import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Product from './Product';
import { getProducts } from '../../../store/products/selectors';

export class ProductsList extends Component {
  render() {
    const { products } = this.props;
    return (
      <div>
        {products.map(product => (
          <Product key={product.id} product={product} />
        ))}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  products: getProducts(state),
});

export default connect(mapStateToProps)(ProductsList);

ProductsList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.object).isRequired,
};
