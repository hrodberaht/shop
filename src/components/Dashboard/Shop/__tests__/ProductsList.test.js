import { ProductsList } from '../ProductsList';

const defaultProps = {
  products: [
    {
      id: 'product-1',
      type: 'Printer',
      name: 'ZX3',
      price: 400,
      inStock: 30,
    },
    {
      id: 'product-2',
      type: 'Printer',
      name: 'ZX5',
      price: 600,
      inStock: 5,
    },
  ],
  handleFetch: jest.fn(),
  token: '12345',
  loaded: false,
};

const setup = buildSetup(ProductsList, defaultProps);

describe('<ProductsList />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render products list if loaded true', () => {
    const { wrapper } = setup();
    wrapper.setProps({ loaded: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleFetch with token', () => {
    const { wrapper } = setup();
    expect(wrapper.instance().props.handleFetch).toHaveBeenCalledWith('12345');
  });
});
