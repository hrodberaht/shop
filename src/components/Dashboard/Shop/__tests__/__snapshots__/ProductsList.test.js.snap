// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsList /> should render products list if loaded true 1`] = `
<div
  className="products-list"
>
  <withRouter(Connect(Product))
    key="product-1"
    product={
      Object {
        "id": "product-1",
        "inStock": 30,
        "name": "ZX3",
        "price": 400,
        "type": "Printer",
      }
    }
  />
  <withRouter(Connect(Product))
    key="product-2"
    product={
      Object {
        "id": "product-2",
        "inStock": 5,
        "name": "ZX5",
        "price": 600,
        "type": "Printer",
      }
    }
  />
</div>
`;

exports[`<ProductsList /> should render without crash 1`] = `
<div
  className="products-list"
>
  <p>
    Loading:
  </p>
</div>
`;
